pronto/api/entry/references.py

```diff
@@ -186,3 +186,169 @@ def unlink_reference(accession, pub_id):
     finally:
         cur.close()
         con.close()
+
+
+@bp.route("/<accession>/references/swissprot/", methods=["PUT"])
+def update_entry_citations(accession):
+    """
+    Retrieve signatures for interpro entry, and the matching proteins, to
+    coordinating updating citations in the interpro oracle db for said proteins.
+
+    :param accession: interpro entry accession
+    """
+    user = auth.get_user()
+    if not user:
+        return jsonify({
+            "status": False,
+            "error": {
+                "title": "Access denied",
+                "message": "Please log in to perform this action."
+            }
+        }), 401
+
+    sig_query = """
+        SELECT METHOD_AC
+        FROM INTERPRO.ENTRY2METHOD
+        WHERE ENTRY_AC = :1
+    """
+
+    con = utils.connect_oracle_auth(user)
+    with con.cursor() as cur:
+        cur.execute(sig_query, [accession])
+        signature_accs = cur.fetchall()
+
+    # process each signature in turn & batch process proteins for each signature
+    signature_accs = [_[0] for _ in signature_accs]
+    invalid_pmids, failed_pub_ids = update_signature_citations(
+        signature_accs, accession, con
+    )
+
+    con.close()
+
+    if len(invalid_pmids) == len(failed_pub_ids) == 0:
+        return jsonify({
+            "status": True,
+        }), 200
+    else:
+        return jsonify({
+            "status": False,
+            "error": {
+                "title": "Failed citation update",
+                "message": f"Failed to update citations and/or relate citations to entry {accession}.",
+                "invalid_pmids": invalid_pmids,
+                "failed_pub_ids": failed_pub_ids,
+            }
+        }), 400
+
+
+def update_signature_citations(
+    sig_accs: list[str],
+    entry_acc: str,
+    orc_con
+) -> tuple[list[str], list[str]]:
+    """Coordinate inserting missing citations from PostGreSQL (e.g. INTTST) db 
+    into the IPRO db (e.g. IPDEV), and relate the new citations to the InterPro entry
+
+    :param sig_accs: list of signature accs associated with interpro entry
+    :param entry_acc: str, IPPRO entry accession
+    :param orc_con: open connection to IPPRO oracle db
+    """
+    invalid_pmids, failed_pub_ids = [], []
+    pmid_query = f"""
+        SELECT P.pubmed_id
```

Comment:
We only need each PubMed ID once:

```suggestion
        SELECT DISTINCT P.pubmed_id
```